yaks/README.md

```diff
@@ -665,11 +678,42 @@ yaks.call(array_of_widgets, mapper: MyCollectionMapper, item_mapper: WidgetMappe
 ```
 
 If the mapper is left unspecified, Yaks will inspect whatever you pass
-it, and try various constant lookups based on naming. These all happen
+it. First it will test the given object against the mapping defined using `mapper_for`.
+If no mapper is found, it wil call `derive_mapper_from_item` or `derive_mapper_from_collection`
+depending on whether the given object is a collection or not. If the object responds
+to `to_ary` it is considered a collection.
+
+Notes:
+* you may override one of these two methods independently from the other.
+* this function takes either a symbol or any instance
```

Comment:
> mapper_for accepts Symbols so that you can specify associations, right?

nope. It is *only* for looking up a specific mapper at the top level, in a case like this

``` ruby
class HomeMapper < Yaks::Mapper
  links 'events', '/api/events'
  links 'shows', '/api/shows'
  links 'users', '/api/users'
end

yaks.call(:home)
```

The `HomeMapper` doesn't care what object you're serializing. You could also call

``` ruby
yaks.call(nil, mapper: HomeMapper)
```

to get the same effect